pkg/compactor/compactor_suite_test.go

```diff
@@ -65,7 +69,7 @@ var _ = SynchronizedBeforeSuite(func() []byte {
 	logger.Infof("Snapshot Directory is: %s", testSnapshotDir)
 
 	// Start the main ETCD process that will run untill all compaction test cases are run
-	etcd, err = utils.StartEmbeddedEtcd(testCtx, testEtcdDir, logger)
+	etcd, err = utils.StartEmbeddedEtcd(testCtx, testEtcdDir, logger, embeddedEtcdPortNo)
```

Comment:
The issue which I faced is that [StartEmbeddedEtcd](https://github.com/gardener/etcd-backup-restore/blob/master/test/utils/utils.go#L46) starting at [random port](https://github.com/gardener/etcd-backup-restore/blob/master/test/utils/utils.go#L54-L57) but AFAIK that port is not exposed to fetch the endpoints correctly. Although `compact pkg` unit test will get passed because they are not throwing an error if defragmentation fails but you look closely you can find the logs.
```
time="2021-11-04T18:03:39+05:30" level=info msg="Starting the defragmentation on etcd leader" suite=compactor
time="2021-11-04T18:03:39+05:30" level=info msg="Defragmenting etcd member[http://localhost:0]" suite=compactor
.
.
.
time="2021-11-04T18:02:24+05:30" level=warning msg="Failed to get status of etcd member[http://localhost:0] with error: context deadline exceeded" suite=compactor
```

And  I got this error while running defragmentation unit tests:
```
{"level":"warn","ts":"2021-11-04T17:22:12.796+0530","caller":"grpclog/grpclog.go:60","msg":"grpc: addrConn.createTransport failed to connect to {http://localhost:0  <nil> 0 <nil>}. Err :connection error: desc = \"transport: Error while dialing dial tcp [::1]:0: connect: can't assign requested address\". Reconnecting..."}
``` 
Amshu suggested me to StartEmbeddedEtcd on particular port, I did that but then `compact pkg` throw an error as it is using to 2 embedded-etcd during testing, then I thought better to make portNo. of StartEmbeddedEtcd configurable.